### PR TITLE
Refactor/Clean leader election related classes

### PR DESCRIPTION
**Pull request by wind57**

### CODE REVIEW

**Review by rohanKanojia — APPROVED** [review]

**Comment by rohanKanojia** [comment]
@wind57 : Usually we require 2 approvals for a pull request to get merged. Most people working on these projects are also working on other projects, so it might take a week or two for pull request to get merged.

File: kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java
Comment (manusa) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java:52]: I'm not sure why this was added on the first place. I need to do some digging before we can merge this.

Comment (shawkins) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java:52]: The go client switch how time was tracked so that it was local to each elector, but we never fully implemented that.

Comment (manusa) [reply kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/extended/leaderelection/LeaderElector.java:52]: I traced it back to the initial implementation but it was never used. I guess I left that there for a follow up.

**Pull request merged**
